nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -81,27 +134,37 @@ def get_conversion_options(self):
 class SpikeGLXLFPInterface(BaseLFPExtractorInterface):
     """Primary data interface class for converting the low-pass (ap) SpikeGLX format."""
 
-    RX = SpikeGLXRecordingExtractor
-
     @classmethod
     def get_source_schema(cls):
-        """Compile input schema for the RecordingExtractor."""
-        source_schema = get_schema_from_method_signature(class_method=cls.__init__, exclude=["x_pitch", "y_pitch"])
-        source_schema["properties"]["file_path"]["description"] = "Path to SpikeGLX file."
-        return source_schema
+        return _get_source_schema(cls)
+
+    def __init__(
+        self,
+        folder_path: FilePathType = None,
+        file_path: FilePathType = None,
```

Comment:
Should not we just keep the `file_path` only for backwards compatibility and extract the folder/directory path from it for so we don't add extra arguments to the function?

Reply:
Since we want to encourage using spike interface, I think using supporting arguments (si uses `folder_path` with `stream_id` to specify the specific file) is better. 
IMOI, If we still want to give users the option to specify the backend, then the recognised arg (`file_path`) should also be present.

Reply:
My argument again it is redundancy. The argument cuts both ways though. We could get `file_path` from `folder_path` or vice-versa. Less sure about the `stream_id`.

In a more general note, in principle "using `spikeinterface` backend" is different from "make our functions call as similar to their recording objects calls". My feeling is that we are concentrating in the former but I am less sure about the latter, maybe am I wrong and we actually want to have similar function calls?

Reply:
yes, we can always get `file_path` given `folder_path` and `stream_id`:
File for spikeglx are of the form `somepath/name_g0_t0.imec0.lf.bin` with `folder_path = "somepath"` and `stream_id="imec0.lf"`. So we can get `file_name` like so:
```python
from pathlib import Path
folder_path = Path(somepath/name_g0_t0.imec0.lf.bin)
stream_id = "imec0.lf"
file_path = list(folder_path.glob(f"*{stream_id}*"))[0]
```
or just specify the `file_path` and get the folder and stream_id like so:
```python
from pathlib import Path
file_path = Path("somepath/name_g0_t0.imec0.lf.bin")
folder_path = str(file_path.parent)
stream_id = "".join(file_path.suffixes[:-1])[1:]  # "imec0.lf"
```
If we are making the design choice to not mimic the args of the spikeinterface/spikeextractors backend, then, IMO the second one where we allow only the `file_path` (like you also mentioned) is better and less confusing for the user. 

I noticed something else: we are not propagating `stream_id` argument in `write_recording` [here](https://github.com/catalystneuro/nwb-conversion-tools/blob/f2ceb69e744ef800d907ebb09668e09037b1ee24/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py#L848) which is used by `add_electrical_series` [here](https://github.com/catalystneuro/nwb-conversion-tools/blob/f2ceb69e744ef800d907ebb09668e09037b1ee24/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py#L454) then why not remove it?

Reply:
Yes, it would be good to get @bendichter and @CodyCBakerPhD thoughts and whether we are should try to emulate the calling signature of `spikeinterface` to some degree. 

Yes, plus is only a single argument which makes it more simple. I feel that many of our functions calls are already to complicated. 

```python
stream_id = "".join(file_path.suffixes[:-1])[1:]  # "imec0.lf"
```
Nice use of suffixes : )

Reply:
> whether we are should try to emulate the calling signature of spikeinterface to some degree.
as discussed with @bendichter lets keep the old arguments: `file_path` instead of `folder_path` + `stream_id`